org-chart/src/contexts/AvatarContext.tsx

```diff
@@ -0,0 +1,105 @@
+    ],
+    'team_korra': [
+        { id:"9", name: 'Asami', skill: 'JUST BADASS', imgSrc: 'https://upload.wikimedia.org/wikipedia/en/thumb/0/02/Asami-neutral-book1.png/220px-Asami-neutral-book1.png'}, 
+        { id:"10", name: 'Korra', skill: 'Water Bender', imgSrc: 'https://decider.com/wp-content/uploads/2020/08/the-legend-of-korra.jpg?quality=75&strip=all'},
+        { id:"11", name: 'Bolin', skill: 'Earth Bender', imgSrc: 'https://a1cf74336522e87f135f-2f21ace9a6cf0052456644b80fa06d4f.ssl.cf2.rackcdn.com/images/characters/large/800/Bolin-The.Legend.of.Korra.webp'},
+        { id:"12", name: 'Mako', skill: 'Fire Bender', imgSrc: 'https://upload.wikimedia.org/wikipedia/en/3/35/Mako_%28The_Legend_of_Korra%29.jpg'}
+    ],
+    'team_kyoshi': [
+        { id:"1", name: 'Kyoshi', skill: 'Earth Bender', nextAvatarTeam: 'team_roku', imgSrc: 'https://img.buzzfeed.com/buzzfeed-static/static/2020-08/13/6/asset/f0d82f7856c0/sub-buzz-3521-1597299837-1.png'},
+    ],
+}
+
+const AVATAR_SEQUENCE = ["team_kyoshi", "team_roku", "team_aang", "team_korra"]
+
+
+type AvatarContextType = {
+    getAvatarTeam: (teamName: string) => Member[],
+    visibleTeams: Set<string>,
+    setVisibleTeams: Dispatch<SetStateAction<Set<string>>>,
+    isTheLastAvatar: (name: string) => boolean,
+    updateVisibleTeams: (name: string) => void
+}
+
+export const AvatarContext = createContext<AvatarContextType>({
+    getAvatarTeam: (teamName: string) => TEAMS[STARTING_TEAM_NAME],
+    visibleTeams: new Set(),
+    setVisibleTeams: (prevState: SetStateAction<Set<string>>) => prevState,
+    isTheLastAvatar: (name: string) => false,
+    updateVisibleTeams: (name: string) => undefined
+})
+
+
+export function AvatarProvider({children}: any) {
+    const [visibleTeams, setVisibleTeams] = useState<Set<string>>(new Set([STARTING_TEAM_NAME])) 
+
+    function getAvatarTeam(teamName: string): Member[] {
+        return TEAMS[teamName]
+    }
+
+    function isTheLastAvatar(name: string): boolean {
+        const teamName = "team_" + name.toLowerCase()
+        return ( Array.from(visibleTeams).pop()  === teamName)
+    }
+
+    function updateVisibleTeams(teamName: string){
+          // find the teams that are in the visible teams but are not supposed to be since they are 
+            // after the last actually visible team
+            let teamIndex = AVATAR_SEQUENCE.length + 1
+            let updatedTeams = new Set(visibleTeams)
+            for (let i = 0; i < AVATAR_SEQUENCE.length; i++) {
+                if (AVATAR_SEQUENCE[i] === teamName) {
+                    teamIndex = i;
+                }
+                if (i > teamIndex) {
+                    updatedTeams.delete(AVATAR_SEQUENCE[i])
+                }
+            }
+            if (updatedTeams !== visibleTeams) {
+               setVisibleTeams(updatedTeams)
+            }
+    }
```

Comment:
The first thing I noticed was the use of a `Set`, which doesn't seem right here. A set is usually to ensure there are no duplicate values, whereas what you really want here is more of a tree structure. A Team would have Members, and a Member can have another Team.

```
const TEAMS = {
    id: 'team_kyoshi',
    members: {
        kyoshi: {
            name: 'Kyoshi',
            skill: 'Earth Bender',
            imgSrc: 'https://.../kyoshi.png'}
            nextTeam: { // this is the "child" of your tree structure
                id: "team_roku",
                members: {
                    sozin: {...},
                    roku: {
                        name: 'Roku',
                        skill: 'Fire Bender',
                        imgSrc: 'https://.../roku.png'}
                        nextTeam: {
                            id: "team_aang",
                            members: { ... }
                        }
                    }
                }
            }
        }
    }
}
```

The nice thing is you don't have to deal with indices, and this structure can have as many team levels as necessary. In fact, multiple members could have their own teams, although not necessary here. You can see it's more flexible and easily extensible.

For the members list of each team, you could choose an array or an object. They have their own advantages, here they're pretty similar since I would just use the object as an array anyways.

```
// ARRAY
members.forEach( (member) => {
  ...
})

// OBJECT
Object.entries(members).forEach( ([key, value]) => {
  // Object.entries returns each key-value pair as an array.
  // key: the "id" of each member, guaranteed to be unique in an object
  // value: the member
})
```

When you want to update team visibility, you start with the leaf node which gets selected. The idea is start with everything being not visible, then traverse up the tree and make each parent visible. In the tree structure above, we only keep track of the next team, but not the previous team. It'd be nice if we had the path of each team, eg. `team_kyoshi/team_roku` or `team_kyoshi/team_roku/team_aang`, so we could easily go back up the tree.

```
function ExampleTeam( {  path, members }:ExampleTeamProps ){
  // eg. path = 'team_kyoshi/team_roku'
  
  return Object.values(members).forEach( (member) => <ExampleMember member={member} path={path} /> )
}

function ExampleMember( { member, path }: ExampleMemberProps ){
  // eg. member is roku
  //     path is still 'team_kyoshi/team_roku`

  const { nextTeam } = member;
  const nextTeamPath = `${path}/${nextTeam.id}` // now we have the path for the next team

  return <ExampleTeam path={nextTeamPath} members={nextTeam.members} />
}
```

Now if you select a member, you can use their path to update visibility:

```
  function updateVisibleTeams(path: string){
    setVisibleTeams( path.split('/') )
  }
  
  // let's say path = 'team_kyoshi/team_roku`
  // updateVisibleTeams(path) => ['team_kyoshi', 'team_roku']

  // and to get the visibility of a team
  function isTeamVisible(teamId: string){
    return visibleTeams.includes(teamId)
  }
```